### ISSUE TITLE
An out-of-the-box Sensu receiver would be useful

### ISSUE THREAD

**Issue by mirath** [flapjack/flapjack#834]
Are there plans to write a Sensu receiver. Could I provide one if it isn't being developed?

**Comment by ali-graham** [comment]
The idea has been floated, but no-one has done any work on one -- that would be useful, thanks. :)

I assume you'll be doing the inverse of https://github.com/sensu/sensu-community-plugins/blob/master/extensions/handlers/flapjack.rb -- building something similar to the Nagios/Icinga receiver on Flapjack's side that hooks into Sensu and pops events onto the redis queue as they're generated?

In what may end up being a Flapjack v3 or later feature we would like to revamp Flapjack's input and output mechanisms, and better document how to integrate with them, but that's a while down the track unfortunately.

**Comment by jessereynolds** [comment]
See also: https://github.com/flapjack/vagrant-flapjack/blob/master/dist/manifests/site.pp (puppet manifest to integrate Sensu and Flapjack)

**Comment by mirath** [comment]
Yes you assume right, the reason I asked is because the handler of the Sensu community connects directly to Redis, so is pretty hard to secure that connection (I could do some port forwards with ssh :shivers:), so I need a receiver that connects directly to Sensu in a secure manner.

There is one thing I don't quite understand. You say that this receiver would end up in a v3 of Flapjack as a revamp of the input and output mechanisms is planned. If that means that my code would get scrapped in such revamp, isn't it possible to include it earlier?. I ask because it would be useful to have it in the default download and package build instead of having to patch my copy of Flapjack every time I do a deploy.

**Comment by ali-graham** [comment]
Yeah, of course you can submit it and we can include it earlier, and we'll make sure the functionality is still supported going forward once it's there -- that was more of an apology for the fact that you currently need to know more about flapjack's internals for this task than you should really need to.
